File: CS-Examples/12_Formulas/RemoveFormulasButKeepValues.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;
using System.Runtime.Remoting.Lifetime;

namespace RemoveFormulasButKeepValues
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a workbook.
			Workbook workbook = new Workbook();

            //Load the file from disk.
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\RemoveFormulasButKeepValues.xlsx");

            //Loop through worksheets.
            foreach (Worksheet sheet in workbook.Worksheets)
            {
                //Loop through cells.
                foreach (CellRange cell in sheet.Range)
                {
                    //If the cell contain formula, get the formula value, clear cell content, and then fill the formula value into the cell.
                    if (cell.HasFormula)
                    {
                        Object value = cell.FormulaValue;
                        cell.Clear(ExcelClearOptions.ClearContent);
                        cell.Value2 = value;
                    }
                }
            }

            // Specify the name for the resulting Excel file
            String result = "Result-RemoveFormulasButKeepValues.xlsx";

            // Save the workbook to the specified file in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: CS-Examples/12_Formulas/RemoveFormulasButKeepValues.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace RemoveFormulasButKeepValues
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "RemoveFormulasButKeepValues.xlsx" from a specific path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\RemoveFormulasButKeepValues.xlsx")

            ' Iterate through each Worksheet in the Workbook
            For Each sheet As Worksheet In workbook.Worksheets

                ' Iterate through each CellRange in the current Worksheet
                For Each cell As CellRange In sheet.Range

                    ' Check if the cell contains a formula
                    If cell.HasFormula Then

                        ' Store the value calculated by the formula in a variable
                        Dim value As Object = cell.FormulaValue

                        ' Clear the content of the cell, removing the formula
                        cell.Clear(ExcelClearOptions.ClearContent)

                        ' Set the value of the cell to the stored value, retaining only the calculated result
                        cell.Value2 = value
                    End If
                Next cell
            Next sheet

            ' Specify the name for the resulting file after removing formulas but keeping values
            Dim result As String = "Result-RemoveFormulasButKeepValues.xlsx"

            ' Save the modified Workbook to a file with Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
